File: Strivers/src/basicsmaths/armstrongnumber/ArmstrongNumberOrNot.java
package basicsmaths.armstrongnumber;

import java.util.Scanner;

public class ArmstrongNumberOrNot {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int number = sc.nextInt();
        System.out.println(armstrongNumberOrNot(number));
    }

    public static boolean armstrongNumberOrNot(int number) {
        int dupNumber = number;
        double total = 0;
        int numberOfDigits = (int) (Math.log10(number) + 1);
        while(dupNumber != 0) {
            int lastDigit = dupNumber % 10;
            total += Math.pow(lastDigit, numberOfDigits);
            dupNumber /= 10;
        }
        return number == (int)total;
    }
}

File: Strivers/src/basicsmaths/armstrongnumber/ArmstrongNumberOrNot.kt
package basicsmaths.armstrongnumber

import java.util.Scanner

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val number = sc.nextInt()
    println(armstrongNumberOrNot(number))
}

fun armstrongNumberOrNot(number: Int): Boolean {
    var numberDup = number
    val numberOfDigits = (Math.log10(numberDup.toDouble()) + 1).toInt()
    var total = 0.0
    while (numberDup != 0) {
        val lastDigit = numberDup % 10
        numberDup /= 10
        total += Math.pow(lastDigit.toDouble(), numberOfDigits.toDouble())
    }
    return number == total.toInt()
}